examples_utils/benchmarks/run_benchmarks.py

```diff
@@ -228,35 +212,49 @@ def run_benchmark_variant(
 
     # Infer examples, SDK and venv path for this benchmark
     args = infer_paths(args, benchmark_dict)
-
     logger.info(f"Datasets directory: '{os.getenv('DATASETS_DIR')}'")
 
-    # Detect if benchmark requires instances running (not just compiling) on
-    # other hosts, and then prepare hosts
-    poprun_hostnames = get_poprun_hosts(cmd)
-    is_distributed = len(poprun_hostnames) > 1 and not args.compile_only
-    if is_distributed:
-        # Setup temporary filesystems on all hosts and modify cmd to use this
-        setup_distributed_filesystems(args, poprun_hostnames)
-
-    start_time = datetime.now()
+    # Detect if a requirements file has been provided
     reqs = benchmark_dict.get("requirements_file")
-    if reqs:
-        logger.info(f"Install python requirements")
-        if not Path(reqs).exists():
-            raise FileNotFoundError(f"Invalid python requirements where specified at {reqs}")
-        subprocess.check_output([sys.executable, "-m", "pip", "install", "-r", str(reqs)])
+    if reqs and not Path(reqs).exists():
+        raise FileNotFoundError(f"Invalid python requirements where specified at {reqs}")
 
+    # Check if poprun is being used
+    poprun_config = get_poprun_config(args, cmd)
+
+    # only validate user supplied hosts if not submitting on SLURM
+    # Similarly, only install requirements if not submitting on SLURM
+    if not args.submit_on_slurm:
+        # Detect if benchmark requires instances running (not just compiling) on
+        # other hosts, and then prepare hosts
+        poprun_hostnames = get_local_poprun_hosts(poprun_config)
+        is_distributed = len(poprun_hostnames) > 1 and not args.compile_only
+
+        if is_distributed and not args.submit_on_slurm:
```

Comment:
Why is second condition in this line, surely this condition can only be reached if not slurm already?